client/rest-high-level/src/main/java/org/elasticsearch/client/indices/IndexTemplateMetaData.java

```diff
@@ -0,0 +1,347 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client.indices;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.cluster.metadata.AliasMetaData;
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.cluster.metadata.MappingMetaData;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.collect.ImmutableOpenMap;
+import org.elasticsearch.common.collect.MapBuilder;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.set.Sets;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.index.mapper.MapperService;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public class IndexTemplateMetaData  {
+
+
+    private final String name;
+
+    private final int order;
+
+    /**
+     * The version is an arbitrary number managed by the user so that they can easily and quickly verify the existence of a given template.
+     * Expected usage:
+     * <pre><code>
+     * PUT /_template/my_template
+     * {
+     *   "index_patterns": ["my_index-*"],
+     *   "mappings": { ... },
+     *   "version": 1
+     * }
+     * </code></pre>
+     * Then, some process from the user can occasionally verify that the template exists with the appropriate version without having to
+     * check the template's content:
+     * <pre><code>
+     * GET /_template/my_template?filter_path=*.version
+     * </code></pre>
+     */
+    @Nullable
+    private final Integer version;
+
+    private final List<String> patterns;
+
+    private final Settings settings;
+
+    private final MappingMetaData mappings;
+
+    private final ImmutableOpenMap<String, AliasMetaData> aliases;
+
+    public IndexTemplateMetaData(String name, int order, Integer version,
+                                 List<String> patterns, Settings settings,
+                                 MappingMetaData mappings,
+                                 ImmutableOpenMap<String, AliasMetaData> aliases) {
+        if (patterns == null || patterns.isEmpty()) {
+            throw new IllegalArgumentException("Index patterns must not be null or empty; got " + patterns);
+        }
+        this.name = name;
+        this.order = order;
+        this.version = version;
+        this.patterns= patterns;
+        this.settings = settings;
+        this.mappings = mappings;
+        this.aliases = aliases;
+    }
+
+    public String name() {
+        return this.name;
+    }
+
+    public int order() {
+        return this.order;
+    }
+
+    public int getOrder() {
```

Comment:
Maybe we should remove these extra getters, and just stick with the noun forms like `order()`? I think we've been trying to standardize on that naming for immutable objects.

Reply:
Did this for the other duplicate getters too